client/src/portal/mod.rs

```diff
@@ -399,4 +415,22 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn change_secret() -> Result<(), Error> {
+        let path = PathBuf::from("../../tests/change_secret.keyring");
+
+        let old_secret = Secret::from(vec![1, 2]);
+        let new_secret = Secret::from(vec![3, 4]);
+        let mut keyring = Keyring::load(&path, old_secret).await?;
+
+        keyring.change_secret(new_secret).await?;
+        keyring.write().await?;
```

Comment:
I might be missing something, but don't this logic (`load` + `write`), also enumerating all items and re-encrypting them, need to be moved to the `change_secret` method; otherwise you will no longer be able to decrypt the items?

Reply:
Agree, just writing again is not enough as a test case anywsays as it would never fail. Instead, I would like to see the test checking that the items before and after changing the secret are still readable.

Reply:
Done, thanks!